src/main/java/minijava/Cli.java

```diff
@@ -267,10 +277,9 @@ static Parameters parse(String... args) {
       Parameters params = new Parameters();
       try {
         new JCommander(params, args);
-        if (params.mainParameters.size() == 1) {
-          params.file = params.mainParameters.get(0);
-        }
+        params.file = params.mainParameters.get(params.mainParameters.size() - 1);
```

Comment:
Really, just go back to the original version where `file` is only set if it is unambiguous and remove [line 278](https://github.com/mj3-16/minijavac/pull/150/files#diff-621504ca28f913dedbeacbc5dfbcea20R278) (which I really don't understand, how is `echo` special?)